### PR TITLE
Implement and use Logger.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: annotations/component-annotations/src/main/java/io/dekorate/component/handler/ComponentHandler.java [hunk 1]
```diff
@@ -17,7 +17,9 @@
 
 import io.dekorate.Configurators;
 import io.dekorate.Handler;
-import io.dekorate.HandlerFactory;
+import io.dekorate.HandlerFactory;import io.dekorate.Logger;
```
Comment (geoand) [inline annotations/component-annotations/src/main/java/io/dekorate/component/handler/ComponentHandler.java]: separate line?

File: annotations/openshift-annotations/src/main/java/io/dekorate/openshift/handler/OpenshiftHandler.java [hunk 2]
```diff
@@ -20,7 +20,10 @@
 import io.dekorate.AbstractKubernetesHandler;
 import io.dekorate.Configurators;
 import io.dekorate.Handler;
-import io.dekorate.HandlerFactory;
+import io.dekorate.HandlerFactory;import io.dekorate.Logger;
```
Comment (geoand) [inline annotations/openshift-annotations/src/main/java/io/dekorate/openshift/handler/OpenshiftHandler.java]: again separate line

File: frameworks/spring-boot/src/main/java/io/dekorate/spring/generator/SpringBootApplicationGenerator.java [hunk 3]
```diff
@@ -15,7 +15,9 @@
  */
 package io.dekorate.spring.generator;
 
-import io.dekorate.Handler;
+import io.dekorate.Handler;import io.dekorate.Logger;
```
Comment (geoand) [inline frameworks/spring-boot/src/main/java/io/dekorate/spring/generator/SpringBootApplicationGenerator.java]: same

**Comment by geoand** [comment]
I must be missing something, so I would be very thankful if you explained why a custom logging abstraction is needed and we can't use sl4f?

**Comment by iocanel** [comment]
@geoand: That's a very good question. Here are some reasons that pop to my mind:

1. Support for ANSI colors.
2. Support for Java annotation processors `Messager`.
3. Potentially easier to integrate with the build tool logging framework (may or may not affect integrations like Quarkus).

**Comment by geoand** [comment]
Sounds good, thanks!

**Pull request merged**